homeassistant/components/bosch_shc/config_flow.py

```diff
@@ -187,16 +187,23 @@ async def async_step_zeroconf(self, discovery_info):
             return self.async_abort(reason="not_bosch_shc")
 
         try:
-            self.info = info = await self._get_info(discovery_info["host"])
+            for host in (
+                discovery_info["host"]
+                if isinstance(discovery_info["host"], list)
+                else [discovery_info["host"]]
+            ):
+                if not host.startswith("169."):  # skip link local address
+                    break
+            self.info = info = await self._get_info(host)
```

Comment:
What is a device only exposes the local link?

Reply:
Actually in this implementation it would still be used. But if both addresses are reported, it will use the non link address.

Reply:
But in that case you still run into the bug you are trying to fix?

Reply:
Sadly yes, would it be better to abort the config flow then?

Reply:
If there is no possible working connection, yeah, abort is the only option.